### PR TITLE
Update SA1513 codefix to use the existing newline character sequence

### LINKED ISSUE #3360: Applying autofixes uses wrong line endings / should honour .editorconfig
`.editorconfig`:
```
[*.cs]
end_of_line = lf
```

`Example.cs`:
```c#
namespace Bug
{
    public static class Example
    {
        public static int MissingNewLine(bool condition)
        {
            if (condition)
            {
                return 42;
            } // SA1513: Closing brace should be followed by a blank line
            return 0;
        }
    }
}
```

Applying the automatic fix for SA1513 (and probably a bunch of others, but this is the one where I tracked down the problem) inserts a CRLF (carriage return + line feed) combination, but the file uses LF only. In Visual Studio, you can observe the state of line endings for the current file in the bottom right corner of the file-editor pane. It should initially say `LF`, when the automatic fix is applied it will switch to `MIXED` instead:
image

The next time the file is opened (but not immediately when the problem is introduced!), Visual Studio will complain about mixed line endings, and suggest to 'normalize' the line endings (using the wrong default option, of-course):
image

The root-cause is probably because [the fixer appears to use a hard-coded CRLF](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/82c2d90907131d939fe396e4191b3a867650615f/StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/LayoutRules/SA1513CodeFixProvider.cs#L66) ([SyntaxFactory.CarriageReturnLineFeed](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.csharp.syntaxfactory.carriagereturnlinefeed?view=roslyn-dotnet-3.9.0)) rather than the currently configured line-end character.

### PR DESCRIPTION
**Pull request by bjornhellander**
Fixes #3360

### CODE REVIEW

**Comment by bjornhellander** [comment]
Did you have something like this in mind, @sharwell?

I can imagine that some other fixers would like to use a new line which is *after* a specified token instead and some probably just have a span to go on. A simpler approach could be to just find the first newline in the document and use that.

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/LayoutRules/SA1513UnitTests.cs [hunk 1]
```diff
@@ -1013,5 +1013,29 @@ public void TestMethod(string extraSupport)
...
+
+        [Fact]
+        [WorkItem(3360, "https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3360")]
+        public async Task TestLineFeedEndOfLinesAsync()
+        {
+            var testCode =
+                "public class TestClass\n" +
+                "{\n" +
+                "}\n" +
+                "// Hello";
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/LayoutRules/SA1513UnitTests.cs:1025]: 💡 You can define a `ReplaceLineEndings` helper like this:
https://github.com/dotnet/roslyn/blob/f64507a8d6b6c3e01f8d1e05ce97552f3c37b341/src/Compilers/Test/Core/FX/StringExtensions.cs

This allows you to use verbatim strings like all the other tests, and just call the extension method to assign the line endings you want.

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/LayoutRules/SA1513UnitTests.cs [hunk 2]
```diff
@@ -1013,5 +1013,29 @@ public void TestMethod(string extraSupport)
...
+            var fixedCode =
+                "public class TestClass\n" +
+                "{\n" +
+                "}\n" +
+                "\n" +
+                "// Hello";
+
+            DiagnosticResult[] expected =
+            {
+                Diagnostic().WithLocation(3, 2),
```
[the remark this answers is not in the document]
Comment (bjornhellander) [reply StyleCop.Analyzers/StyleCop.Analyzers.Test/LayoutRules/SA1513UnitTests.cs:1036]: Will do

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/LayoutRules/SA1513UnitTests.cs (hunk 1 above)
Comment (bjornhellander) [reply StyleCop.Analyzers/StyleCop.Analyzers.Test/LayoutRules/SA1513UnitTests.cs:1025]: Absolutely. Would you like to keep the "fancy" conversions, or just assume CRLF in the input?

**Comment by bjornhellander** [comment]
Updated, @sharwell

**Comment by bjornhellander** [comment]
Any other comments, @sharwell?

File: StyleCop.Analyzers/StyleCop.Analyzers/Helpers/TokenHelper.cs [hunk 3]
```diff
@@ -175,5 +175,35 @@ internal static bool IsFollowedByWhitespace(this SyntaxToken token)
...
+            var leadingTrivia = token.LeadingTrivia;
+            for (var i = leadingTrivia.Count - 1; i >= 0; i--)
+            {
+                if (leadingTrivia[i].IsKind(SyntaxKind.EndOfLineTrivia))
+                {
+                    return leadingTrivia[i];
+                }
+            }
+
+            var prevToken = token.GetPreviousToken();
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/Helpers/TokenHelper.cs:196]: 💡 If there is any way to test this with a preceding `#region` directive, that would be good (preprocessor directives have embedded line endings that are not EndOfLineTrivia). Will leave this for future improvement though.

**Review by sharwell — APPROVED** [review]

File: StyleCop.Analyzers/StyleCop.Analyzers/Helpers/TokenHelper.cs (hunk 3 above)
Comment (bjornhellander) [reply StyleCop.Analyzers/StyleCop.Analyzers/Helpers/TokenHelper.cs:196]: It looks like this rule never triggers when the closing brace is followed by a directive, so I don't see a way to test that case with this rule. But if this method will be used in other code fixers as well, then it seems reasonable to do.

**Pull request merged**